pyogrio/tests/test_geopandas_io.py

```diff
@@ -510,19 +653,21 @@ def test_write_mixed_geometries(tmp_path, driver, ext):
         crs="EPSG:4326",
     )
 
-    filename = tmp_path / f"test.{ext}"
-    write_dataframe(df, filename, driver=driver)
+    filename = tmp_path / f"test{ext}"
+
+    if ext == ".fgb":
+        # For .fgb, spatial_index=False to avoid the rows being reordered
+        write_dataframe(df, filename, spatial_index=False)
+    else:
+        write_dataframe(df, filename)
 
     # Drivers that support mixed geometries will default to "Unknown" geometry type
     assert read_info(filename)["geometry_type"] == "Unknown"
     result = read_dataframe(filename)
-    if driver == "FlatGeobuf":
-        # FlatGeobuf results in mixed row order in case of mixed geometries
-        result = result.sort_values("col").reset_index(drop=True)
     assert_geodataframe_equal(result, df)
```

Comment:
```suggestion
    assert_geodataframe_equal(result, df, check_geom_type=True)
```

(otherwise a corresponding MultiPoint and Point will not be detected if they are "spatially" equal)